File: examples/sample.cpp
#include <cmath>
#include <iostream>

// Sample C++ file to exercise the code renderer.
struct Point {
    double x;
    double y;

    double distance_to(const Point& other) const {
        double dx = x - other.x;
        double dy = y - other.y;
        return std::sqrt(dx * dx + dy * dy);
    }
};

int main() {
    Point a{0, 0};
    Point b{3, 4};
    std::cout << "distance: " << a.distance_to(b) << '\n';
    return 0;
}

File: examples/sample.py
"""Sample Python file to exercise the code renderer."""
from dataclasses import dataclass


@dataclass
class Point:
    x: float
    y: float

    def distance_to(self, other: "Point") -> float:
        return ((self.x - other.x) ** 2 + (self.y - other.y) ** 2) ** 0.5


def main() -> None:
    a, b = Point(0, 0), Point(3, 4)
    print("distance:", a.distance_to(b))


if __name__ == "__main__":
    main()
